dll/win32/msgina/branding.c

```diff
@@ -58,4 +63,261 @@ Brand_Cleanup(
+                              (HMENU)IDC_BAR,
+                              pBrand->hDllInstance,
+                              NULL);
+
+#undef ID2STR
+
+    /* Retrieve the total height of logo + bar */
+    SetRectEmpty(&rcCtrl);
+#if 0
+    GetWindowRect(hwndLogo, &rc);
+    MapWindowPoints(NULL, hWnd, (LPPOINT)&rc, sizeof(RECT)/sizeof(POINT));
+    rcCtrl.bottom += rc.bottom - rc.top;
+
+    if (hwndBar)
+    {
+        GetWindowRect(hwndBar, &rc);
+        MapWindowPoints(NULL, hWnd, (LPPOINT)&rc, sizeof(RECT)/sizeof(POINT));
+        rcCtrl.bottom += rc.bottom - rc.top;
+    }
+#else
+    rcCtrl.bottom += pBrand->LogoHeight;
+    if (hwndBar)
+        rcCtrl.bottom += pBrand->BarHeight;
+#endif
+
+    /* Count the number of children windows in this parent window */
+    for (hwndChild = GetWindow(hWnd, GW_CHILD);
+         hwndChild;
+         hwndChild = GetNextWindow(hwndChild, GW_HWNDNEXT))
+    {
+        /* Skip the logo and bar controls */
+        // LONG_PTR nID = GetWindowLongPtrW(hwndChild, GWLP_ID);
+        // if (nID == IDC_ROSLOGO || nID == IDC_BAR)
+        if (hwndChild == hwndLogo || hwndChild == hwndBar)
+            continue;
+
+        ++nNumChildren;
+    }
+
+    /* Start moving windows around */
+    hdwp = BeginDeferWindowPos(nNumChildren);
+
+    // TODO: Increase height of window by (height_logo + height_bar)
+    // and reposition accordingly.
+    // if (hdwp)
+        // hdwp = DeferWindowPos(hdwp,
+                              // hWnd,
+                              // HWND_TOP,
+                              // rc.left, rcCtrl.bottom + rc.top,
+                              // 0, 0,
+                              // SWP_NOZORDER | SWP_NOACTIVATE);
+    AdjustWindowSize(hWnd, pBrand, rcCtrl.right, rcCtrl.bottom);
+
+    /* Move each child window down */
+    for (hwndChild = GetWindow(hWnd, GW_CHILD);
+         hwndChild;
+         hwndChild = GetNextWindow(hwndChild, GW_HWNDNEXT))
+    {
+        /* Skip the logo and bar controls */
+        // LONG_PTR nID = GetWindowLongPtrW(hwndChild, GWLP_ID);
+        // if (nID == IDC_ROSLOGO || nID == IDC_BAR)
+        if (hwndChild == hwndLogo || hwndChild == hwndBar)
+            continue;
+
+        GetWindowRect(hwndChild, &rc);
+        MapWindowPoints(HWND_DESKTOP /*NULL*/, hWnd, (LPPOINT)&rc, sizeof(RECT)/sizeof(POINT));
+
+        if (hdwp)
+            hdwp = DeferWindowPos(hdwp,
+                                  hwndChild,
+                                  HWND_TOP,
+                                  rc.left, rcCtrl.bottom + rc.top,
+                                  0, 0,
+                                  SWP_NOSIZE | SWP_NOZORDER | SWP_NOACTIVATE);
+    }
+
+    if (hdwp)
+        EndDeferWindowPos(hdwp);
+}
+
+
+#if 0
+VOID
+Brand_AnimateBar(
+    _In_ PDLG_DATA pDlgData,
+    _In_ UINT uMsg,
+    _In_ LPARAM lParam)
+{
+    if (!pDlgData->hBarBitmap)
+        return;
+
+    switch (uMsg)
+    {
+#if 0
+        case WM_INITDIALOG:
```

Comment:
Oh no, the bar animation thing is still being defined out, even with nested `#if 0`'s

Reply:
Yes, the working code is still in gui.c . I was here thinking whether I could move it out too.